src/utilities/quaternions.jl

```diff
@@ -34,7 +34,7 @@ end
 @inline Base.getindex(x::Quaternion, i::Integer) = x.data[i]
 Base.isapprox(x::Quaternion, y::Quaternion) = all((x.data .≈ y.data))
 
-function qrotation(axis::StaticVector{3}, theta::Real)
+function qrotation(axis::StaticVector{3}, theta)
```

Comment:
This seems to be the only real change needed here, right? Can we leave it at `Number`?

Reply:
That is probably fine. But what is the benefit of the `Number` type annotation? Either you can take the `sin`\`cos` of the type, or you can't, and you'll get an error too.

Don't know why you would want to here, but you can take the `sin` or square matrices, for example. I am generally in favor of unconstraining the types, so long as it does not hinder error reporting or dispatch. In this case, there's no dispatch needed, since there's only one definition of the `qrotation`method.

Reply:
after working with a lot of other people's code, I started following the rule to annotate types for documentation,.... Makes it so much easier to get started with a code base ;) And here, we even have the perfect type to annotate it with ;)

Reply:
Perfect until someone invents a non-`Number` type that represents an angle! But that's unlikely.

I agree generally about type annotations helping as documentation, but not sure what else a user could think `theta` is in this case. But I think it does no harm to add `::Number`. I'll do that!

Reply:
Thank you :) It's a rule of thumb I like to follow in the code base wherever I can ... And if someone actually runs into a problem, it's very easy to remove the type constraint ;)

Reply:
Seems resolved by https://github.com/JuliaPlots/AbstractPlotting.jl/pull/129/commits/7d41fe03778ba2f246724d34a1dd0a2a547c8b0a